nifi-nar-bundles/nifi-jolt-record-bundle/nifi-jolt-record-processors/src/main/java/org/apache/nifi/processors/jolt/record/JoltTransformRecord.java

```diff
@@ -0,0 +1,534 @@
+            .build();
+
+    private final static List<PropertyDescriptor> properties;
+    private final static Set<Relationship> relationships;
+    private volatile ClassLoader customClassLoader;
+    private final static String DEFAULT_CHARSET = "UTF-8";
+
+    // Cache is guarded by synchronizing on 'this'.
+    private volatile int maxTransformsToCache = 10;
+    private final Map<String, JoltTransform> transformCache = new LinkedHashMap<String, JoltTransform>() {
+        @Override
+        protected boolean removeEldestEntry(Map.Entry<String, JoltTransform> eldest) {
+            final boolean evict = size() > maxTransformsToCache;
+            if (evict) {
+                getLogger().debug("Removing Jolt Transform from cache because cache is full");
+            }
+            return evict;
+        }
+    };
+
+    static {
+        final List<PropertyDescriptor> _properties = new ArrayList<>();
+        _properties.add(RECORD_READER);
+        _properties.add(RECORD_WRITER);
+        _properties.add(JOLT_TRANSFORM);
+        _properties.add(CUSTOM_CLASS);
+        _properties.add(MODULES);
+        _properties.add(JOLT_SPEC);
+        _properties.add(TRANSFORM_STRATEGY);
+        _properties.add(TRANSFORM_CACHE_SIZE);
+        properties = Collections.unmodifiableList(_properties);
+
+        final Set<Relationship> _relationships = new HashSet<>();
+        _relationships.add(REL_SUCCESS);
+        _relationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(_relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        final List<ValidationResult> results = new ArrayList<>(super.customValidate(validationContext));
+        final String transform = validationContext.getProperty(JOLT_TRANSFORM).getValue();
+        final String customTransform = validationContext.getProperty(CUSTOM_CLASS).getValue();
+        final String modulePath = validationContext.getProperty(MODULES).isSet() ? validationContext.getProperty(MODULES).getValue() : null;
+
+        if (!validationContext.getProperty(JOLT_SPEC).isSet() || StringUtils.isEmpty(validationContext.getProperty(JOLT_SPEC).getValue())) {
+            if (!SORTR.getValue().equals(transform)) {
+                final String message = "A specification is required for this transformation";
+                results.add(new ValidationResult.Builder().valid(false)
+                        .explanation(message)
+                        .build());
+            }
+        } else {
+            final ClassLoader customClassLoader;
+
+            try {
+                if (modulePath != null) {
```

Comment:
I don't believe we need to be dealing with ClassLoaders anymore. If the Property Descriptor is marked with `.dynamicallyModifiesClassPath` I think we're all set - the framework will handle this for you.

Reply:
This is copy/paste from the original Jolt processor :) I'll replace it as you said and make sure all is well.